keras/engine/training.py

```diff
@@ -1396,10 +1396,8 @@ def _standardize_user_data(self, x, y,
 
         output_shapes = []
         for output_shape, loss_fn in zip(self._feed_output_shapes, self._feed_loss_fns):
-            if loss_fn.__name__ == 'sparse_categorical_crossentropy':
+            if loss_fn is losses.sparse_categorical_crossentropy:
                 output_shapes.append(output_shape[:-1] + (1,))
-            elif getattr(losses, loss_fn.__name__, None) is None:
```

Comment:
I have my custom loss function that accepts data of the same shape as `sparse_categorical_crossentropy` which doesn't match output shape of model. It seems this line was needed to skip output shape check for custom loss functions, because currently it fails as they don't match.

Reply:
I was on the hunt for this bug too, and found your comment. I am seeing the exact same thing. My output is shaped `(None, None, 4)` and my target output is shaped `(None, None, 5)` (the last column defines if the values belong to background/foreground, which are treated differently when computing the loss). After removing this line, this difference in shape is not allowed, giving the error:
```
ValueError: Error when checking target: expected regression to have shape (None, None, 4) but got array with shape (1, 102438, 5)
```

@frexvahi @fchollet can this be hotfixed?

Reply:
This is a significant regression which breaks a lot of my code too.